addons/base_automation/models/base_automation.py

```diff
@@ -117,6 +118,23 @@ def _onchange_state(self):
                 ),
             }}
 
+    @api.constrains('filter_pre_domain', 'filter_domain', 'model_id')
+    def _check_filter_domain(self):
+        for action in self.sudo():
+            ctx = action._get_eval_context()
+            if action.filter_pre_domain:
+                try:
+                    domain = safe_eval.safe_eval(action.filter_pre_domain, ctx)
```

Comment:
avoid making sql queries just for validation.
We have a piece of code that does just that at afdb546e4448e8aab170f9221fe1e4ee3aea2cd1

Also, it needs a test

Reply:
@odony @rco-odoo a long time ago I have write a POC to create an env with readonly database connection : https://github.com/odoo/odoo/pull/85290, maybe we can add this feature to replace the explain.

@kebeclibre I have add test and update the code.